src/plugins/intel_cpu/src/nodes/random_uniform.cpp

```diff
@@ -399,135 +528,220 @@ std::pair<uint64_t, uint64_t> RandomUniform::computePhilox(void* out, size_t out
+                                bfloat16 min,
+                                bfloat16 range,
+                                bfloat16* out,
+                                size_t el_to_copy) {
+    const auto mask = static_cast<uint32_t>((1UL << 8) - 1);
+    const auto divisor = static_cast<float>(1) / (1UL << 8);
 
-    switch (m_output_prc) {
-        EL_CASE(f32)
-        EL_CASE(f16)
-        EL_CASE(bf16)
-        EL_CASE(i32)
-        EL_CASE(i64)
-        EL_CASE(f64)
-        default:
-            THROW_CPU_NODE_ERR("has unsupported output precision: ", m_output_prc);
+    for (size_t i = 0lu; i < el_to_copy; i++) {
+        const float ret = (in[i] & mask) * divisor;
+        out[i] = ret * range + min;
     }
+}
 
-#undef EL_CASE
+inline void convertToOutputTypeMersenne(const uint32_t* in,
+                                int32_t min,
+                                int32_t range,
+                                int32_t* out,
+                                size_t el_to_copy) {
+    for (size_t i = 0lu; i < el_to_copy; i++) {
+        out[i] = static_cast<int32_t>(in[i] % range + min);
+    }
 }
 
-std::string RandomUniform::getPrimitiveDescriptorType() const {
-    auto selectedPrimitiveDesc = getSelectedPrimitiveDescriptor();
+inline void convertToOutputTypeMersenne(const uint32_t* in,
+                                int64_t min,
+                                int64_t range,
+                                int64_t* out,
+                                size_t el_to_copy) {
+    for (size_t i = 0lu; i < el_to_copy; i++) {
+        out[i] = static_cast<int64_t>(((static_cast<uint64_t>(in[i * 2]) << 32) + in[i * 2 + 1]) % range + min);
+    }
+}
+} // namespace
 
-    impl_desc_type type = impl_desc_type::undef;
-    if (selectedPrimitiveDesc) {
-        type = selectedPrimitiveDesc->getImplementationType();
+void RandomUniform::computeMersenneTwister(void* out, size_t out_el_num) {
+    // When both seed values are equal to zero RandomUniform should generate non-deterministic sequence.
+    if (m_global_seed == 0lu && m_op_seed == 0lu) {
+        std::srand(static_cast<unsigned int>(std::time(nullptr)));
+        m_global_seed = std::rand();
     }
 
-    std::string str_type;
+    uint32_t mersenne_state[MERSENNE_STATE_N];
 
-    auto add_type = [&](std::string t) {
-        if (!str_type.empty() && t.c_str()[0] != '_')
-            str_type += "_";
-        str_type += t;
-    };
+    auto out_u8 = reinterpret_cast<uint8_t*>(out);
 
-#define SEARCH_TYPE(_type)                                          \
-    if ((type & impl_desc_type::_type) == impl_desc_type::_type)    \
-        add_type(#_type)
+    if (m_jit_kernel) {
+#if defined(OPENVINO_ARCH_X86_64)
+        parallel_nt(m_threads_num, [&](const int ithr, const int nthr) {
+                auto& p = m_thread_params[ithr];
+                if (p.work_amount == 0lu) {
+                    return;
+                }
+                // auto n = p.n_shift;
```

Comment:
Could you please remove commented code lines?

Reply:
Done!